modules/server/src/network/aggregations/AggregationAccumulator.ts

```diff
@@ -37,13 +47,19 @@ const resolveAggregations = ({ data, accumulator, requestedFields }: ResolveAggr
 		const { aggregations, hits } = data;
 
 		const isFieldAvailable = !!aggregations[requestedField];
-		const type = ALL_NETWORK_AGGREGATION_TYPES_MAP.get(requestedField);
+
+		/*
+		 * requested field will always be in ALL_NETWORK_AGGREGATION_TYPES_MAP
+		 * GQL schema validation will throw an error earlier if a requested field isn't in the schema
+		 */
+		const type = ALL_NETWORK_AGGREGATION_TYPES_MAP.get(requestedField) as SupportedAggregation;
 		const existingAggregation = accumulator[requestedField];
 
 		if (isFieldAvailable) {
 			accumulator[requestedField] = addToAccumulator({
 				existingAggregation,
-				aggregation: aggregations[requestedField],
+				// similarly due to GQL schema validation this will not be undefined
+				aggregation: aggregations[requestedField]!,
 				type,
```

Comment:
neither `isFieldAvailable` variable or directly putting a conditional of `if(requestedField in aggregations)` works to assert not undefined